composer/utils/fx_utils.py

```diff
@@ -123,16 +149,106 @@ def detect_residual_pattern(gm: GraphModule):
     raise NotImplementedError('detect_residual_pattern is currently not implemented.')
 
 
-def replace_residual_with_stochastic(gm: GraphModule):
-    """Replaces residual pattern with their stoachstic equivalent.
+def _get_ancestors(node: Node) -> List[Node]:
+    ancestorNodes = []
+    while node.op != 'placeholder':
+        ancestorNodes.append(node)
+        node = node.all_input_nodes[0]
+    return ancestorNodes
+
+
+def _get_residual_path(nodeLHS: Node, nodeRHS: Node) -> Tuple[List[Node], List[Node]]:
+    """Walk backwards from nodeLHS and nodeRSH to the root and construct lists of their parents.
 
     Arguments:
-        gm (GraphModule): The source FX-traced graph.
+        nodeLHS (Node): left-hand side node for a binary operator
+        nodeRHS (Node): right-hand side node for a binary operator
 
     Returns:
-        GraphModule: Modified GraphModule.
+        (lhsAncestors, rhsAncesotors): Two lists of nodes containing ancestors for ``nodeLHS`` and ``nodeRHS`` with
+            their common ancestors removed.
     """
-    raise NotImplementedError('replace_residual_with_stochastic is currently not implemented.')
+    lhsAncestors = _get_ancestors(nodeLHS)
+    rhsAncesotors = _get_ancestors(nodeRHS)
+
+    # Iterate from back and eliminate common nodes
+    while lhsAncestors and rhsAncesotors and lhsAncestors[-1] == rhsAncesotors[-1]:  # type: ignore
```

Comment:
curious about the need for a type ignore?

Reply:
It's throwing the following error. Not sure what's the issue. 
 
  /workdisk/daya/composer/composer/utils/fx_utils.py:186:45 - error: Operator "==" not supported for types "Node | Unbound" and "Node | Unbound"
    Operator "==" not supported for types "Unbound" and "Unbound" when expected type is "List[Node]" (reportGeneralTypeIssues)

Reply:
Ah, it's probably coming from the lines below...instead of using `del` syntax, could you do `lhsAncestors.pop()`

Reply:
Thanks. That fixed it. pyright is weird.